services/core-api/app/api/EMLI_contacts/resources/EMLI_contact_list.py

```diff
@@ -0,0 +1,104 @@
+from app.api.EMLI_contacts.models.EMLI_contact_type import EMLIContactType
+from app.api.EMLI_contacts.response_models import EMLI_CONTACT_MODEL
+
+
+class EMLIContactListResource(Resource, UserMixin):
+    parser = CustomReqparser()
+    parser.add_argument(
+        'emli_contact_type_code',
+        type=str,
+        trim=True,
+        help='EMLI contact type code',
+        location='json')
+    parser.add_argument(
+        'mine_region_code', type=str, trim=True, help='EMLI mine region code', location='json')
+    parser.add_argument('first_name', type=str, trim=True, help='EMLI First name', location='json')
+    parser.add_argument('last_name', type=str, trim=True, help='EMLI Last name.', location='json')
+    parser.add_argument('email', type=str, help='EMLI email.', required=True, location='json')
+    parser.add_argument(
+        'phone_number', type=str, help='EMLI phone number', required=True, location='json')
+    parser.add_argument(
+        'fax_number', type=str, help='EMLI Regional Office fax number', location='json')
+    parser.add_argument(
+        'mailing_address_line_1',
+        type=str,
+        help='EMLI Regional Office mailing address line 1',
+        location='json')
+    parser.add_argument(
+        'mailing_address_line_2',
+        type=str,
+        help='EMLI Regional Office mailing address line 2',
+        location='json')
+    parser.add_argument(
+        'is_major_mine', type=bool, help='is major mine contact? true/false', location='json')
+    parser.add_argument(
+        'is_general_contact', type=bool, help='is is_general_contact? true/false', location='json')
+    parser.add_argument(
+        'deleted_ind', type=bool, help='Deleted indicator: true/false', location='json')
+
+    @api.doc(
+        params={'is_major_mine': 'EMLI user is related to a major mine? true/false'},
+        description='Returns all EMLI contacts information.')
+    @api.marshal_with(EMLI_CONTACT_MODEL, code=201, envelope='records')
+    @requires_any_of([VIEW_ALL, MINESPACE_PROPONENT])
+    def get(self, mine_region_code=None):
+        is_major_mine = request.args.get('is_major_mine') == 'true'
+
+        if mine_region_code:
+            return EMLIContact.find_EMLI_contacts_by_mine_region(mine_region_code, is_major_mine)
+
+        return EMLIContact.get_all(is_major_mine)
+
+    @api.doc(description='Create a EMLI contact.')
+    @api.expect(parser)
+    @api.marshal_with(EMLI_CONTACT_MODEL, code=200)
+    @requires_role_edit_emli_contacts
+    def post(self):
+        data = self.parser.parse_args()
+
+        contact_type = data.get('emli_contact_type_code', None)
+        contact_desc = EMLIContactType.find_contact_type(contact_type)
+
+        general_contacts = EMLIContact.find_EMLI_general_contacts()
+        mmo_contact = EMLIContact.find_EMLI_contact('MMO')
+        roe_contact = EMLIContact.find_EMLI_contact('ROE', data.mine_region_code)
+
+        unique_by_region = roe_contact.emli_contact_type_code
```

Comment:
Just for thought and I don't think it is an issue here but this type of validation can open this up to potential race conditions or simultaneous posts